src/components/Profile/Setting.vue

```diff
@@ -0,0 +1,209 @@
+<template>
+  <b-container class="no-gutters pl-0 pr-0" fluid>
+    <div id="avatar-container">
+      <b-avatar size="9.375rem" :src="photoURL" class="mb-4" />
+      <p id="avatar-upload-text">上傳</p>
+    </div>
+    <div class="section">
+      <h4>顯示名稱</h4>
+      <div class="text-box">
+        <span class="m-0 flex-grow-1">{{ displayName }}</span>
+        <span class="text-box-divider" />
+        <b-button
+          id="rename-btn"
+          v-b-modal.rename-modal
+          variant="white"
+          size="lg"
+          class="m-0 font-size-1"
+        >更改</b-button>
+        <b-modal
+          id="rename-modal"
+          centered
+          title="更改顯示名稱"
+          title-class="font-weight-bold"
+          header-class="rename-modal-header border-0"
+          footer-class="border-0"
+        >
+          <b-container>
+            <b-form>
+              <b-form-group>
+                <template #description>
+                  <p class="mt-3">
+                    顯示名稱在一個月內僅可以更改兩次。<br>
+                    且更改後，將會影響你過去所編輯過文章的名稱。
+                  </p>
+                </template>
+                <b-form-input id="input-username" size="lg" required />
+              </b-form-group>
+            </b-form>
+          </b-container>
+          <template #modal-footer="{ ok, cancel }">
+            <b-button
+              class="rename-modal-btn"
+              variant="white"
+              type="button"
+              @click="cancel()"
+            >取消</b-button>
+            <b-button
+              id="btn-ok"
+              class="rename-modal-btn"
+              variant="white"
+              type="submit"
+              @click="ok()"
+            >儲存</b-button>
+          </template>
+        </b-modal>
+      </div>
+    </div>
+    <div class="section pb-3">
+      <h4>註冊信箱</h4>
+      <div class="text-box bg-light">
+        <span class="m-0 flex-grow-1">{{ email }}</span>
+      </div>
+    </div>
+    <div class="section pt-5">
+      <h4>偏好設定</h4>
+      <PreferenceItem
+        v-for="preference in mockPreference"
+        :key="preference"
```

Comment:
The keys are using non-primitive value, it may cause the update error. 
```suggestion
        v-for="(preference, preferenceIndex) in mockPreference"
        :key="preferenceIndex"
```